tests/test_file_storage.py

```diff
@@ -0,0 +1,407 @@
+import json
+import os.path
+
+import pytest
+from dotenv import load_dotenv
+
+from unstract.sdk.exceptions import FileStorageError
+from unstract.sdk.file_storage.fs_impl import FileStorage
+from unstract.sdk.file_storage.fs_permanent import PermanentFileStorage
+from unstract.sdk.file_storage.fs_provider import FileStorageProvider
+
+load_dotenv()
+
+
+class TEST_CONSTANTS:
+    READ_FOLDER_PATH = os.environ.get("READ_FOLDER_PATH")
+    WRITE_FOLDER_PATH = os.environ.get("WRITE_FOLDER_PATH")
+    READ_PDF_FILE = os.environ.get("READ_PDF_FILE")
+    READ_TEXT_FILE = os.environ.get("READ_TEXT_FILE")
+    WRITE_PDF_FILE = os.environ.get("WRITE_PDF_FILE")
+    WRITE_TEXT_FILE = os.environ.get("WRITE_TEXT_FILE")
+    TEST_FOLDER = os.environ.get("TEST_FOLDER")
+    GCS_BUCKET = os.environ.get("GCS_BUCKET")
+    TEXT_CONTENT = os.environ.get("TEXT_CONTENT")
+    FILE_STORAGE_ENV = "FILE_STORAGE"
+
+
+def file_storage(provider: FileStorageProvider):
+    credentials = json.loads(os.environ.get(TEST_CONSTANTS.FILE_STORAGE_ENV))
+    file_storage = FileStorage(provider=provider, credentials=credentials)
+    assert file_storage is not None
+    return file_storage
+
+
+def permanent_file_storage(provider: FileStorageProvider):
+    credentials = json.loads(os.environ.get(TEST_CONSTANTS.FILE_STORAGE_ENV))
+    file_storage = PermanentFileStorage(provider=provider, credentials=credentials)
+    assert file_storage is not None
+    return file_storage
```

Comment:
NIT: You can add these with a `pytest.fixture` decorator in `conftest.py` to reuse them in your tests rather than explicitly creating them in `paramterize()`. You can refer [these tests](https://github.com/Zipstack/llm-whisperer-python-client/blob/c97cb4582d9dade85788e894f4060d5cec100b31/tests/conftest.py#L10) for reference

Reply:
I tried that. However my case is slightly different where I need to call the fixture itself with different values. When I tried doing that by keeping the file_storage and permanent_file_storage as fixtures, I see an error. I am not investing much time here for now as this is  improvisation. Will come back to it when I get a chance.